lib/query-builder.js

```diff
@@ -3139,6 +3139,80 @@ function TermDef(textQuery) {
   }
 }
 util.inherits(TermDef, QueryDef);
+
+function before() {
+   var args = mlutil.asArray.apply(null, arguments);
+   if(args.length === 0 || args[0] === null)
+     throw new Error('missing Timestamp');
+   if(args.length>1)
+     throw new Error('Only one timestamp is allowed');
+   var arg = new String(args[0]);
+
+   if(!(typeof(args[0]) == 'string')) {
+     if(typeof(args[0]) == 'number') {
+      arg = new String(args[0]);
+     }
+     else if(args[0] instanceof Date) {
+      var epochMilli = args[0].getTime();
+      arg = new String(epochMilli);
+     }
+     else {
+      throw new Error('The timestamp argument needs to be either String, Number or Date');
+     }
+   }
+  return new BeforeQueryDef(new TimestampDef(arg));
+}
+/** @ignore */
+function BeforeQueryDef(timestampQuery) {
+  if (!(this instanceof BeforeQueryDef)) {
+    return new BeforeQueryDef(timestampQuery);
+  }
+
+  if (timestampQuery instanceof TimestampDef) {
+    this['before-query'] = timestampQuery;
+  } else {
+    throw new Error('missing query');
```

Comment:
Would the error be more easily understood as 'missing timestamp value for query' both here and in AfterQuery?